### ISSUE TITLE
Get error: x86_64 GS register is not set as expected. EOS VM OC may not run correctly on this platform

### ISSUE THREAD

**Issue by matthewdarwin** [AntelopeIO/leap#1216]
Occasionally nodeos will issue error message "x86_64 GS register is not set as expected. EOS VM OC may not run correctly on this platform".  Seems to happen on next restart after kill -9 of nodeos.

Nodeos issues this message after already loading state (in heap mode) and then exits.  Our systemd unit will restart nodeos and then it runs fine.

What's going on here?  Why doesn't it check this before loading all the state?  Wasting cycles loading RAM and then unloading RAM. What is this test doing?

**Comment by spoonincode** [comment]
> Nodeos issues this message after already loading state (in heap mode) and then exits

This message is probably a red herring then as it's only a informational warning and doesn't cause nodeos to exit per se. When you see this error what kind of exit code does nodeos exit with?

**Comment by matthewdarwin** [comment]
Full log files have been sent privately.

![image](https://github.com/AntelopeIO/leap/assets/26171839/674d6fef-d591-428b-8f6e-290cd66c8f15)

```
May 25 18:16:15 waxtest-ship11 nodeos[429697]: info  2023-05-25T18:16:15.366 nodeos    http_plugin.cpp:378           add_async_handler    ] add api url: /v1/net/connect
May 25 18:16:15 waxtest-ship11 nodeos[429697]: info  2023-05-25T18:16:15.366 nodeos    http_plugin.cpp:378           add_async_handler    ] add api url: /v1/net/connections
May 25 18:16:15 waxtest-ship11 nodeos[429697]: info  2023-05-25T18:16:15.366 nodeos    http_plugin.cpp:378           add_async_handler    ] add api url: /v1/net/disconnect
May 25 18:16:15 waxtest-ship11 nodeos[429697]: info  2023-05-25T18:16:15.366 nodeos    http_plugin.cpp:378           add_async_handler    ] add api url: /v1/net/status
May 25 18:16:15 waxtest-ship11 nodeos[429697]: info  2023-05-25T18:16:15.366 nodeos    producer_plugin.cpp:1222      plugin_startup       ] producer plugin:  plugin_startup() begin
May 25 18:16:19 waxtest-ship11 nodeos[429697]: warn  2023-05-25T18:16:19.003 read-0    executor.cpp:142              executor             ] x86_64 GS register is not set as expected. EOS VM OC may not run correctly on this platform
May 25 18:16:19 waxtest-ship11 nodeos[429697]: warn  2023-05-25T18:16:19.079 read-1    executor.cpp:142              executor             ] x86_64 GS register is not set as expected. EOS VM OC may not run correctly on this platform
May 25 18:16:19 waxtest-ship11 nodeos[429697]: info  2023-05-25T18:16:19.157 nodeos    producer_plugin.cpp:1316      plugin_shutdown      ] exit shutdown
May 25 18:16:19 waxtest-ship11 nodeos[429697]: info  2023-05-25T18:16:19.157 nodeos    net_plugin.cpp:3908           plugin_shutdown      ] shutdown..
May 25 18:16:19 waxtest-ship11 nodeos[429697]: info  2023-05-25T18:16:19.157 nodeos    net_plugin.cpp:2722           plugin_shutdown      ] close 0 connections
May 25 18:16:19 waxtest-ship11 nodeos[429697]: warn  2023-05-25T18:16:19.158 read-2    executor.cpp:142              executor             ] x86_64 GS register is not set as expected. EOS VM OC may not run correctly on this platform
May 25 18:16:19 waxtest-ship11 nodeos[429697]: info  2023-05-25T18:16:19.195 nodeos    net_plugin.cpp:3912           plugin_shutdown      ] exit shutdown
May 25 18:16:19 waxtest-ship11 nodeos[429697]: CHAINBASE: Writing "state" database file, this could take a moment...
May 25 18:16:20 waxtest-ship11 nodeos[429697]: CHAINBASE: Writing "state" database file, 3% complete...
May 25 18:16:49 waxtest-ship11 nodeos[429697]: CHAINBASE: Writing "state" database file, 89% complete...
May 25 18:16:49 waxtest-ship11 nodeos[429697]: CHAINBASE: Writing "state" database file, flushing buffers...
May 25 18:17:14 waxtest-ship11 nodeos[429697]: CHAINBASE: Writing "state" database file, complete.
May 25 18:17:16 waxtest-ship11 nodeos[429697]: info  2023-05-25T18:17:16.282 nodeos    http_plugin.cpp:368           plugin_shutdown      ] exit shutdown
May 25 18:17:16 waxtest-ship11 nodeos[429697]: error 2023-05-25T18:17:16.284 nodeos    main.cpp:161                  main                 ] 3170000 producer_exception: Producer exception
May 25 18:17:16 waxtest-ship11 nodeos[429697]: read-only threads failed to start. num_threads_started: 0, time_slept_ms: 1000ms
May 25 18:17:16 waxtest-ship11 nodeos[429697]:     {"n":0,"t":1000}
May 25 18:17:16 waxtest-ship11 nodeos[429697]:     nodeos  producer_plugin.cpp:1283 plugin_startup
May 25 18:17:16 waxtest-ship11 nodeos[429697]:     {}
May 25 18:17:16 waxtest-ship11 nodeos[429697]:     nodeos  producer_plugin.cpp:1296 plugin_startup
May 25 18:17:16 waxtest-ship11 systemd[1]: nodeos.service: Main process exited, code=exited, status=254/n/a
```

**Comment by bhazzard** [comment]
We believe that this is likely a red herring that will be addressed by #1223
